cmd/crossplane/core/core.go

```diff
@@ -204,24 +206,34 @@ func (c *startCommand) Run(s *runtime.Scheme, log logging.Logger) error { //noli
 			log.Info("Beta feature enabled", "flag", features.EnableBetaCompositionFunctionsExtraResources)
 		}
 
-		clienttls, err := certificates.LoadMTLSConfig(
-			filepath.Join(c.TLSClientCertsDir, initializer.SecretKeyCACert),
-			filepath.Join(c.TLSClientCertsDir, corev1.TLSCertKey),
-			filepath.Join(c.TLSClientCertsDir, corev1.TLSPrivateKeyKey),
-			false)
-		if err != nil {
-			return errors.Wrap(err, "cannot load client TLS certificates")
-		}
-
 		m := xfn.NewMetrics()
 		metrics.Registry.MustRegister(m)
 
-		// We want all XR controllers to share the same gRPC clients.
-		functionRunner = xfn.NewPackagedFunctionRunner(mgr.GetClient(),
+		fnRunnerOpts := []xfn.PackagedFunctionRunnerOption{
 			xfn.WithLogger(log),
-			xfn.WithTLSConfig(clienttls),
 			xfn.WithInterceptorCreators(m),
-		)
+		}
+
+		if c.TLSAllowInsecure {
```

Comment:
Is it possible to avoid this if/else branch? e.g. Using a helper function that takes all the TLS options and produces a `*tls.Config`?

Reply:
Thanks for the hint. That is indeed a better way to handle it. I changed it.